diff --git a/lib/main.js b/lib/main.js
index 79d973c22f06403bfee45ff122bd384841c30da8..9392de262140f8f336146bef2419794532e4170f 100644
--- a/lib/main.js
+++ b/lib/main.js
@@ -69,9 +69,10 @@ function expand (config) {
     )
   }
 
-  for (const processKey in config.parsed) {
-    environment[processKey] = config.parsed[processKey]
-  }
+  // PATCH: don't write to process.env
+  // for (const processKey in config.parsed) {
+  //   environment[processKey] = config.parsed[processKey]
+  // }
 
   return config
 }