diff --git a/dist/client.mjs b/dist/client.mjs
index 3d769de748044f6eb25e70d5803169c3d6cfc889..943aceabe74dadf96107b8965c9b65522ae2c050 100644
--- a/dist/client.mjs
+++ b/dist/client.mjs
@@ -309,6 +309,11 @@ class ViteNodeRunner {
 			__vite_ssr_dynamic_import__: request,
 			__vite_ssr_exports__: exports,
 			__vite_ssr_exportAll__: (obj) => exportAll(exports, obj),
+			__vite_ssr_exportName__: (name, getter) => Object.defineProperty(exports, name, {
+        enumerable: true,
+        configurable: true,
+        get: getter,
+      }),
 			__vite_ssr_import_meta__: meta,
 			require: createRequire(href),
 			exports: cjsExports,
